aconcli/service/auth.go

```diff
@@ -0,0 +1,101 @@
+package service
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+const (
+	UserRuntimeDir = "/run/user"
+	AuthTableFile  = "tokens.json"
+)
+
+// map vmid to associated access token
+type AuthTable map[string]string
+
+var authTableMutex sync.Mutex
+
+func getUserAuthTable(uid string) (AuthTable, error) {
+	records, err := os.ReadFile(filepath.Join(UserRuntimeDir, uid, AuthTableFile))
+	if err != nil {
+		return nil, err
+	}
+	var authTable AuthTable
+	if err := json.Unmarshal(records, &authTable); err != nil {
+		return nil, err
+	}
+	return authTable, nil
+}
+
+func GetAuthToken(uid string, vmid string) (string, error) {
+	authTableMutex.Lock()
```

Comment:
What if there are multiple instances of `aconcli` racing on updating this table? Guess a (advisory) file lock is needed here instead of a mutex.

Reply:
Using syscall.Flock to prevent accesses from multiple `aconcli` processes